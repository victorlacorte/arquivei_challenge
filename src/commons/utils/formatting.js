function toBRL(number) {
  return number.toLocaleString('pt-BR', { style: 'currency', currency: 'BRL' });
}

export {
  toBRL,
};
