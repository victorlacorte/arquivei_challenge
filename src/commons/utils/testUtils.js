/* eslint-disable react/jsx-filename-extension */
/* eslint-disable react/prop-types */
import React from 'react';
import { ThemeProvider } from 'styled-components';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount, shallow } from 'enzyme';

import defaultTheme from 'commons/styles/theme';

function mountWithTheme(child) {
  return mount(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

function shallowWithTheme(child) {
  return shallow(child, {
    wrappingComponent: ({ children }) => (
      <ThemeProvider theme={defaultTheme}>
        {children}
      </ThemeProvider>
    ),
  });
}

export {
  mountWithTheme,
  shallowWithTheme,
};
