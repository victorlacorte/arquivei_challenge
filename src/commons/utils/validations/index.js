export { default as cnpjValidations } from './cnpj';
export { default as consultKeysValidations } from './consultKeys';
export { default as creditCardValidations } from './creditCard';
export { default as formValidations } from './formValidations';
