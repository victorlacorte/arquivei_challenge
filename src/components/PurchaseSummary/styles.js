import styled, { css } from 'styled-components';

const Receipt = styled.div`
  padding: 24px;
  background-color: ${({ theme }) => theme.palette.grey100.main};
  border-style: solid;
  border-width: 0 0 10px 0;
  border-image-source: url("/images/receipt-border.png");
  border-image-slice: 30;
  border-image-width: 1;
  border-image-outset: 0;
  border-image-repeat: stretch;
  max-height: 304px;
`;

const Title = styled.h2`
  text-align: center;
  margin-bottom: 32px;

  ${({ theme }) => css`
      ${theme.typography.h2};
  `}
`;

const Hr = styled.hr`
  border-style: dashed;
  border-color: ${({ theme }) => theme.palette.grey100.dark};
  margin-bottom: 24px;
  width: 100%;
`;

// TODO set Table's max-width
const Table = styled.table`
  width: 100%;
  table-layout: fixed;
  border-collapse: collapse;
  margin-bottom: 24px;

  & th {
    color: ${({ theme }) => theme.palette.grey400.main};
    font-family: ${({ theme }) => theme.typography.fontFamily.Rubik};
  }

  & td {
    font-family: ${({ theme }) => theme.typography.fontFamily.Roboto};
    text-align: center;
  }
`;

const OriginalTotal = styled.div`
  color: ${({ theme }) => theme.palette.grey400.main};
  font-family: ${({ theme }) => theme.typography.fontFamily.Roboto};

  & span {
    margin-left: 8px;
    text-decoration: line-through;
  }
`;

const DiscountedTotal = styled.div`
  font-family: ${({ theme }) => theme.typography.fontFamily.Roboto};
  font-size: 20px;
  font-weight: 900;
  color: ${({ theme }) => theme.palette.success.main};

  & span:not(:first-child) {
    margin-left: 8px;
  }
`;

const Total = styled.div`
  font-family: ${({ theme }) => theme.typography.fontFamily.Roboto};
  font-size: 20px;
  font-weight: 900;
`;

export {
  Receipt,
  Title,
  Hr,
  Table,
  OriginalTotal,
  DiscountedTotal,
  Total,
};
