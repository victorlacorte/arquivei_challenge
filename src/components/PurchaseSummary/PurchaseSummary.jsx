import React from 'react';
import PropTypes from 'prop-types';

import { toBRL } from 'commons/utils/formatting';

import {
  DiscountedTotal,
  Hr,
  OriginalTotal,
  Receipt,
  Table,
  Title,
  Total,
} from './styles';

function PurchaseSummary(props) {
  const {
    consultKeys,
    keyUnitPrice,
    total,
    discountedTotal,
    percentageDiscount,
  } = props;

  return (
    <Receipt>
      <Title>
        Detalhes da compra
      </Title>

      <Table>
        <thead>
          <tr>
            <th># Chaves</th>
            <th>Valor unit√°rio</th>
          </tr>
        </thead>

        <tbody>
          <tr>
            <td>{Number(consultKeys).toFixed(0)}</td>
            <td>
              {toBRL(keyUnitPrice)}
            </td>
          </tr>
        </tbody>
      </Table>

      <Hr />

      <Title>
        Total
      </Title>

      {discountedTotal
        ? (
          <>
            <OriginalTotal>
              De:
              <span>
                {toBRL(total)}
              </span>
            </OriginalTotal>

            <DiscountedTotal>
              <span>
                {toBRL(discountedTotal)}
              </span>

              <span>
                (
                {Number(percentageDiscount).toFixed(0)}
                % off)
              </span>
            </DiscountedTotal>
          </>
        )
        : (
          <Total>
            {toBRL(total)}
          </Total>
        )}
    </Receipt>
  );
}

PurchaseSummary.defaultProps = {
  consultKeys: 0,

  keyUnitPrice: 0,

  total: 0,

  discountedTotal: undefined,

  percentageDiscount: undefined,
};

PurchaseSummary.propTypes = {
  consultKeys: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),

  keyUnitPrice: PropTypes.PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),

  total: PropTypes.PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),

  discountedTotal: PropTypes.PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),

  percentageDiscount: PropTypes.PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
};

export default PurchaseSummary;
