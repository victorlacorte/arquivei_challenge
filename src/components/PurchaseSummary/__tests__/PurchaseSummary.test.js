import React from 'react';

import PurchaseSummary from 'components/PurchaseSummary';
import { shallowWithTheme } from 'commons/utils/testUtils';

describe('<PurchaseSummary />', () => {
  it('creates a PurchaseSummary component', () => {
    const wrapper = shallowWithTheme(<PurchaseSummary />);

    // Default values have no discount
    expect(wrapper.find('styles__DiscountedTotal').children()).toHaveLength(0);

    expect(wrapper).toMatchSnapshot();
  });

  it('creates a PurchaseSummary component with total and discountedTotal props', () => {
    const wrapper = shallowWithTheme(<PurchaseSummary total={10} discountedTotal={8} />);

    expect(wrapper.find('styles__OriginalTotal').children()).toHaveLength(2);
    expect(wrapper.find('styles__DiscountedTotal').children()).toHaveLength(2);
  });
});
