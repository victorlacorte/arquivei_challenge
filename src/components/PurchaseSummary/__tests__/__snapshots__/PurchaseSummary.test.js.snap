// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PurchaseSummary /> creates a PurchaseSummary component 1`] = `
<styles__Receipt>
  <styles__Title>
    Detalhes da compra
  </styles__Title>
  <styles__Table>
    <thead>
      <tr>
        <th>
          # Chaves
        </th>
        <th>
          Valor unitário
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          0
        </td>
        <td>
          R$ 0,00
        </td>
      </tr>
    </tbody>
  </styles__Table>
  <styles__Hr />
  <styles__Title>
    Total
  </styles__Title>
  <styles__Total>
    R$ 0,00
  </styles__Total>
</styles__Receipt>
`;
